mm2src/coins/hd_wallet/pubkey.rs

```diff
@@ -87,20 +26,23 @@ pub trait ExtractExtendedPubkey {
         XPubExtractor: HDXPubExtractor + Send;
 }
 
+/// A trait for extracting an extended public key from an external source.
 #[async_trait]
 pub trait HDXPubExtractor: Sync {
-    async fn extract_utxo_xpub(
+    async fn extract_xpub(
         &self,
-        trezor_utxo_coin: String,
+        trezor_coin: String,
```

Comment:
I guess this trait was not designed for trezor only so this `trezor_coin` probably will be wrapped in a enum when we add another device

Reply:
Exactly. It should be refactored when adding support for other devices.